woocommerce/woocommerce
# Added wc_sanitize_phone_number() to remove invalid characters.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR introduces a fix for sanitizing phone numbers in the checkout process by utilizing wc_sanitize_phone_number() before validation. This resolves issues where Unicode control characters (e.g., left-to-right markers) would otherwise be included in the submitted data, leading to validation failures for otherwise valid phone numbers.

The addition of wc_sanitize_phone_number() ensures that any unintended, non-visible Unicode characters are removed, improving the reliability and accuracy of phone number validation in WooCommerce.

Closes #52012



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Shortcode Checkout and input a phone number containing Unicode control characters (e.g., copy a number with LTR markers like this one `‪‪+31123456789‬` ).
2. Submit the checkout and confirm the validation accepts the phone number without errors.
3. Check that regular phone numbers are also validated and processed as expected without impacting the checkout experience.
4. Repeat steps above for Blocks Checkout



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Prevented phone numbers containing Unicode control characters from failing validation during Checkout.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
